lib/puppet/provider/cpdb_create/cpdb_create.rb

```diff
@@ -0,0 +1,38 @@
+Puppet::Type.type(:cpdb_create).provide(:cpdb_create) do
+
+  commands :cpdb => '/usr/share/candlepin/cpdb'
+
+  def create
+    create_database
+    write_done_file
+  end
+
+  def exists?
+    File.exist?(done_file)
+  end
+
+  private
+
+  def create_database
+    cpdb(
+      "--create",
+      "--schema-only",
+      "--dbhost=#{resource[:db_host]}",
+      "--dbport=#{resource[:db_port]}",
+      "--database=#{resource[:db_name]}#{resource[:ssl_options]}",
+      "--user=#{resource[:db_user]}",
+      "--password=#{resource[:db_password]}"
+    )
+  end
+
+  def done_file
+    "/var/lib/candlepin/cpdb_create_done"
```

Comment:
Perhaps this should be a hidden file
```suggestion
    "/var/lib/candlepin/.puppet-candlepin-cpdb_create_done"
```